src/proto/streams/streams.rs

```diff
@@ -352,6 +362,15 @@ impl<B> StreamRef<B>
         })
     }
 
+    pub fn is_recv_empty(&self) -> bool {
+        let mut me = self.inner.lock().unwrap();
+        let me = &mut *me;
+
+        let stream = me.store.resolve(self.key);
+
+        stream.state.is_recv_closed() && stream.pending_recv.is_empty()
```

Comment:
Actually, i think this is okay if it's only meant to determine whether eos may be set...